File: src/com/programmingfire/java/Game.java
package com.programmingfire.java;

import java.util.Random;
import java.util.Scanner;

public class Game {
    private final int randomNumber;
    private final Random random = new Random();
    private final Scanner scanner = new Scanner(System.in);
    private int guesses = 0;

    public Game() {
        randomNumber = random.nextInt(1, 100);
    }
    public Game(int origin, int bound) {
        randomNumber = random.nextInt(origin, bound);
    }

    public void play() {
        while (true) {
            var guess = takeInput();
            var isWon = decide(guess);
            if (isWon) {
                break;
            }
        }
    }

    private int takeInput() {
        System.out.print("Enter The Number: ");
        var guess = scanner.nextInt();
        guesses++;
        return guess;
    }

    private boolean decide(int guess) {
        if (guess == randomNumber) {
            System.out.println("You Won The Game In " + guesses + " Guesses!");
            return true;
        } else if (guess > randomNumber) {
            System.out.println("Your Number Is Larger Than The Random Number");
        } else {
            System.out.println("Your Number Is Smaller Than The Random Number");
        }

        return false;
    }
}

File: src/com/programmingfire/kotlin/Game.kt
package com.programmingfire.kotlin

import java.util.*

class Game {
    private val randomNumber: Int
    private val random = Random()
    private val scanner = Scanner(System.`in`)
    private var guesses: Int = 0

    constructor() {
        randomNumber = random.nextInt(1, 100)
    }

    constructor(origin: Int, bound: Int) {
        randomNumber = random.nextInt(origin, bound)
    }

    fun play() {
        while (true) {
            val guess = takeInput()
            val isWon = decide(guess)
            if (isWon) {
                break
            }
        }
    }

    private fun takeInput(): Int {
        print("Enter The Number: ")
        val guess = scanner.nextInt()
        guesses++
        return guess
    }

    private fun decide(guess: Int): Boolean {
        if (guess == randomNumber) {
            println("You Won The Game In $guesses Guesses!")
            return true
        } else if (guess > randomNumber) {
            println("Your Number Is Larger Than The Random Number")
        } else {
            println("Your Number Is Smaller Than The Random Number")
        }
        return false
    }
}